packages/xo-server/src/xo-mixins/resource-sets.js

```diff
@@ -174,6 +178,27 @@ export default class {
       set.name = name
     }
     if (subjects) {
+      await Promise.all(
+        difference(set.subjects, subjects).map(async subjectId =>
+          Promise.all(
+            (await this._xo.getAclsForSubject(subjectId)).map(async acl => {
+              try {
+                const object = this._xo.getObject(acl.object)
+                if (object.type === 'VM' && object.resourceSet === id) {
+                  await this._xo.removeAcl(subjectId, acl.object, acl.action)
+                  $defer.onFailure(() =>
+                    this._xo.addAcl(subjectId, acl.object, acl.action)
+                  )
+                }
+              } catch (err) {
+                if (!startsWith(err && err.message, 'no such object')) {
```

Comment:
Use [`noSuchObject.is`](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-common/src/api-errors.js#L35).